src/cc65/assignment.c

```diff
@@ -54,6 +54,69 @@
 
 
 
+static int CopyStruct (ExprDesc* LExpr, ExprDesc* RExpr)
+/* Copy the struct represented by RExpr to the one represented by LExpr */
+{
+    /* If the size is that of a basic type (char, int, long), we will copy
+    ** the struct using the primary register, otherwise we use memcpy. In
+    ** the former case, push the address only if really needed.
+    */
+    const Type* ltype  = LExpr->Type;
+    const Type* stype  = GetReplacementType (ltype);
+    int         UseReg = (stype != LExpr->Type);
```

Comment:
The source expression `(stype != ltype)` looks cleaner than the above expression.

Reply:
Done together with other changes.